tests/test_api.py

```diff
@@ -85,3 +87,23 @@ def test_qasm_to_phir(self) -> None:
         """
 
         assert qasm_to_phir(qasm, QtmMachine.H1_1)
+
+    def test_pytket_with_wasm(self) -> None:
+        this_dir = Path(Path(__file__).resolve()).parent
+        w = WasmFileHandler(f"{this_dir}/data/wasm/testfile.wasm")
+
+        c = Circuit(6, 6)
+        c0 = c.add_c_register("c0", 3)
+        c1 = c.add_c_register("c1", 4)
+        c2 = c.add_c_register("c2", 5)
+
+        c.add_wasm_to_reg("multi", w, [c0, c1], [c2])
+        c.add_wasm_to_reg("add_one", w, [c2], [c2])
+        c.add_wasm_to_reg("no_return", w, [c2], [])
+        c.add_wasm_to_reg("no_parameters", w, [], [c2])
+
+        c.add_wasm_to_reg("add_one", w, [c0], [c0], condition=c1[0])
+
+        phir = pytket_to_phir(c, QtmMachine.H1_1)
+
+        assert phir
```

Comment:
Could we add fine-grained tests here? Specifically, it will be nice to ensure the issues we identified in the chat last month about args and returns don't occur.

Reply:
Can you remind me of what issues those were?